src/ts/garagePage/getCars.ts

```diff
@@ -0,0 +1,109 @@
+import { carSVG } from "../data/carSVG";
+import { fetchGarage } from "./fetchGarage";
+import { Car } from "../types";
+import { updateCar } from "./updateCar";
+import { addMainPage } from "../routing/pages/mainPage";
+
+export async function getCars():Promise<string> {
+  fetchGarage().then((response: Array<Car>) => {
+    const garage = document.querySelector('.garage');
+    response.forEach((item: Car) => {
+      const car = document.createElement('div');
+      car.classList.add('car');
+      
+      car.append(addCarInfo(item), addCarTrack(item));
+      garage?.append(car)
+    })
+  });
+
+  return 'ok'
+}
+
+function addCarInfo(item: Car):HTMLElement {
+  const carInfo = document.createElement('div');
+  carInfo.classList.add('car__info');
+
+  const carID = document.createElement('span');
+  carID.innerText = item.id.toString();
+  carID.classList.add('car_id');
+  carID.style.display = 'none';
+
+  const carButtons = document.createElement('span');
+  carButtons.classList.add('car-buttons');
+  const buttonsWrapper = document.createElement('div');
+  buttonsWrapper.classList.add('buttons-wrapper');
+  const fButton = document.createElement('button');
+  fButton.className = ('button button_blue');
+  fButton.innerText = 'Select';
+  fButton.addEventListener('click', (e: Event) => {
+    const nameInput = document.querySelector('.update-car .car__name') as HTMLInputElement;
+    nameInput.classList.remove('disabled');
+    nameInput.value = item.name;
+
+    const colorInput = document.querySelector('.update-car .car__color') as HTMLInputElement;
+    colorInput.classList.remove('disabled');
+    colorInput.value = item.color;
+
+    const buttonInput = document.querySelector('.update-car .car__button') as HTMLInputElement;
+    buttonInput.classList.remove('disabled');
+    buttonInput.after(carID);
+    buttonInput.addEventListener('click', updateCar)
+  });
+
+  const sButton = document.createElement('button');
```

Comment:
I'm bot quite sure if usage of Promises is ok in addEventListener, it's better to make a separate function